flaskapp/views/ivrflow.py

```diff
@@ -420,17 +418,36 @@ def get_profile():
     else:
         return message
 
+
 # http://127.0.0.1:5000/new_user?username=testuser&&type=patient&&timezone=US/Pacific&&calltime=5:30:00&&phone=123-456-789
 def new_user():
     all_args = request.args.to_dict()
     rec1 = User.create(**all_args)
     rec1.save()
-    return {"success" : 200, "newuser" : model_to_dict(rec1)}
+    return {"success": 200, "newuser": model_to_dict(rec1)}
+
 
 def unsubscribe():
     user_info = request.args.to_dict()
     ph = user_info.get('phone')
-    del_row  = User.delete().where(User.phone == ph)
+    del_row = User.delete().where(User.phone == ph)
     if del_row > 0:
         return {"success": 200, "message": "user unsubscribed"}
-    return {"message": "user not found", "failed": 400}
\ No newline at end of file
+    return {"message": "user not found", "failed": 400}
+
+
+# getting set of from_ numbers from twilio call logs  that are less than 7 days old
+def get_from_phone_number_from_twilio_call_logs():
+    account_sid = os.environ['TWILIO_ACCOUNT_SID']
+    auth_token = os.environ['TWILIO_AUTH_TOKEN']
+    client = Client(account_sid, auth_token)
+
+    start = datetime.datetime.now()
+    calls = [call.from_ for call in client.calls.list() if
+             ((start.date() - call.start_time.date())
+              + (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
+                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
+                                      seconds=call.start_time.time().second))) < datetime.timedelta(days=7)
```

Comment:
7 is a magic number here; do we need to put it to the settings.py file?

Reply:
done!